### PR TITLE
OCaml files

### PR DESCRIPTION
**Pull request by mmottl**
These patches make OCaml-related Vim-files more consistent with the OCaml-community repository at: https://github.com/ocaml/vim-ocaml

### CODE REVIEW

File: runtime/indent/ocaml.vim [hunk 1]
```diff
@@ -101,7 +102,7 @@ function! GetOCamlIndent()
 
  " Return double 'shiftwidth' after lines matching:
  if lline =~ '^\s*|.*->\s*$'
-   return ind + 2 * shiftwidth()
+   return ind + &sw + &sw
```
Comment (k-takata) [inline runtime/indent/ocaml.vim]: This change looks unintended.

File: runtime/indent/ocaml.vim [hunk 2]
```diff
@@ -172,7 +173,7 @@ function! GetOCamlIndent()
  " Indent if current line begins with 'and':
  elseif line =~ '^\s*and\>'
    if lline !~ '^\s*\(and\|let\|type\)\>\|\<end\s*$'
-     return ind - shiftwidth()
+     return ind - &sw
```
Comment (k-takata) [inline runtime/indent/ocaml.vim]: same as above

File: runtime/indent/ocaml.vim [hunk 3]
```diff
@@ -199,14 +200,14 @@ function! GetOCamlIndent()
  " or 'method':
  elseif line =~ '^\s*\(constraint\|inherit\|initializer\|method\)\>'
    if lline !~ s:obj
-     return indent(search('\<\(object\|object\s*(.*)\)\s*$', 'bW')) + shiftwidth()
+     return indent(search('\<\(object\|object\s*(.*)\)\s*$', 'bW')) + &sw
```
Comment (k-takata) [inline runtime/indent/ocaml.vim]: same as above

File: runtime/indent/ocaml.vim [hunk 4]
```diff
@@ -199,14 +200,14 @@ function! GetOCamlIndent()
...
    endif
 
  endif
 
  " Add a 'shiftwidth' after lines ending with:
  if lline =~ '\(:\|=\|->\|<-\|(\|\[\|{\|{<\|\[|\|\[<\|\<\(begin\|do\|else\|fun\|function\|functor\|if\|initializer\|object\|parser\|private\|sig\|struct\|then\|try\)\|\<object\s*(.*)\)\s*$'
-   let ind = ind + shiftwidth()
+   let ind = ind + &sw
```
Comment (k-takata) [inline runtime/indent/ocaml.vim]: same as above

File: runtime/indent/ocaml.vim [hunk 5]
```diff
@@ -263,7 +264,7 @@ function! GetOCamlIndent()
 
  " Subtract a 'shiftwidth' after lines matching 'match ... with parser':
  if lline =~ '\<match\>.*\<with\>\s*\<parser\s*$'
-   let ind = ind - shiftwidth()
+   let ind = ind - &sw
```
Comment (k-takata) [inline runtime/indent/ocaml.vim]: same as above

File: runtime/indent/ocaml.vim [hunk 6]
```diff
@@ -3,8 +3,8 @@
 " Maintainers:  Jean-Francois Yuen   <jfyuen@happycoders.org>
 "               Mike Leary           <leary@nwlink.com>
 "               Markus Mottl         <markus.mottl@gmail.com>
-" URL:          http://www.ocaml.info/vim/indent/ocaml.vim
-" Last Change:  2017 Jun 13
+" URL:          https://github.com/ocaml/vim-ocaml
+" Last Change:  2013 Jun 29
```
Comment (k-takata) [inline runtime/indent/ocaml.vim]: It seems that the file becomes older.

File: runtime/indent/ocaml.vim (hunk 2 above)
Comment (mmottl) [reply runtime/indent/ocaml.vim]: Fixed

File: runtime/indent/ocaml.vim (hunk 1 above)
Comment (mmottl) [reply runtime/indent/ocaml.vim]: Fixed

File: runtime/indent/ocaml.vim (hunk 3 above)
Comment (mmottl) [reply runtime/indent/ocaml.vim]: Fixed

File: runtime/indent/ocaml.vim (hunk 4 above)
Comment (mmottl) [reply runtime/indent/ocaml.vim]: Fixed

File: runtime/indent/ocaml.vim (hunk 5 above)
Comment (mmottl) [reply runtime/indent/ocaml.vim]: Fixed

File: runtime/indent/ocaml.vim (hunk 6 above)
Comment (mmottl) [reply runtime/indent/ocaml.vim]: Fixed

**Comment by brammool** [comment]
Markus Mottl wrote:

> These patches make OCaml-related Vim-files more consistent with the
> OCaml-community repository at: https://github.com/ocaml/vim-ocaml

Thanks.  I'll include the runtime file changes and make a patch for the
filetype detection.

In the dune indent file, the last command should be "setlocal".
I'll make that change, please also do that in your version.

-- 
hundred-and-one symptoms of being an internet addict:
72. Somebody at IRC just mentioned a way to obtain full motion video without
    a PC using a wireless protocol called NTSC, you wonder how you never
    heard about it

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by brammool** [comment]
> These patches make OCaml-related Vim-files more consistent with the
> OCaml-community repository at: https://github.com/ocaml/vim-ocaml

I found one problem: syntax/ocaml.vim is now using line continuation,
therefore it needs to set and restore 'cpo'.


-- 
From "know your smileys":
 y:-)	Bad toupee

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by mmottl** [comment]
@brammool Thanks, I have merged your changes into the OCaml repository.

**Pull request closed without merging**